kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/config/VirtualClusterTest.java

```diff
@@ -106,4 +114,49 @@ void disallowsGatewaysWithDuplicateNames() {
                 .isInstanceOf(IllegalConfigurationException.class)
                 .hasMessageContaining("Gateway names for a virtual cluster must be unique. The following gateway names are duplicated: [dup]");
     }
+
+    public static Stream<Arguments> rejectVirtualClusterNamesThatArentDnsLabels() {
+        return Stream.of(argumentSet("hyphen at start", "-cluster"),
+                argumentSet("hyphen at end", "cluster-"),
+                argumentSet("hyphen at start and end", "-cluster-"),
+                argumentSet("too long", Strings.repeat("a", 64)));
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void rejectVirtualClusterNamesThatArentDnsLabels(String clusterName) {
+        // Given
+        var gateways = List.of(new VirtualClusterGateway("mygateway1", portIdentifiesNode1, null, Optional.empty()));
+
+        // When
+        // Then
+        assertThatThrownBy(() -> {
+            new VirtualCluster(clusterName, targetCluster, null, null, gateways, false, false, NO_FILTERS);
+        }).isInstanceOf(IllegalConfigurationException.class)
+                .hasMessageContaining("Virtual cluster name '" + clusterName
+                        + "' is invalid. It must be less than 64 characters long and match pattern ^[a-z0-9]([-a-z0-9]*[a-z0-9])?$ (case insensitive)");
+    }
+
+    public static Stream<Arguments> acceptVirtualClusterNamesThatAreDnsLabels() {
+        return Stream.of(argumentSet("start with number", "1a"),
+                argumentSet("start with alpha", "a1"),
+                argumentSet("start with uppercase alpha", "A1"),
```

Comment:
By extension when we need to do a case insensitive unique name check otherwise the DNS is going to screw up later. Though its likely people have far bigger problems if they have two virtual cluster names that only differ in case.